ideascube/serveradmin/tests/test_catalog.py

```diff
@@ -30,21 +35,27 @@
     ],
     ids=[
         'foo',
+        'utf8',
         'missing-id',
         'missing-name',
         'missing-url',
     ])
 def input_file(tmpdir, request):
     path = tmpdir.join('foo.yml')
 
+    lines = []
+
     if 'id' in request.param:
-        path.write('id: {id}\n'.format(**request.param), mode='a')
+        lines.append('id: {id}\n'.format(**request.param))
 
     if 'name' in request.param:
-        path.write('name: {name}\n'.format(**request.param), mode='a')
+        lines.append('name: {name}\n'.format(**request.param))
 
     if 'url' in request.param:
-        path.write('url: {url}'.format(**request.param), mode='a')
+        lines.append('url: "{url}"'.format(**request.param))
+
+    path.write_text('\n'.join(lines), encoding='utf-8')
```

Comment:
line in `lines` already have a `\n`.
Either we just `''.join(...)` or we remove the `\n` in the `lines.append` before.